subprojects/gst-plugins-good/gst/rtp/gstrtph266pay.c

```diff
@@ -0,0 +1,792 @@
+
+  gst_rtp_buffer_unmap (&rtp);
+  GST_DEBUG_OBJECT (rtph266pay, "Pushing UP %" GST_PTR_FORMAT, out_buf);
+  return gst_rtp_base_payload_push (rtpbasepay, out_buf);
+error:
+  gst_buffer_unref (out_buf);
+  return GST_FLOW_ERROR;
+}
+
+static GstFlowReturn
+gst_rtp_h266_pay_push_fu (GstRtpH266Pay * rtph266pay, const Nalu * nalu)
+{
+  GstRTPBasePayload *rtpbasepay = GST_RTP_BASE_PAYLOAD (rtph266pay);
+  GstBufferList *out_buflist = gst_buffer_list_new ();
+
+  // TODO: Consider DONL
+  guint fu_max_size = GST_RTP_BASE_PAYLOAD_MTU (rtpbasepay) - FU_HDR_LEN;
+  guint fu_pyl_max_size = gst_rtp_buffer_calc_payload_len (fu_max_size, 0, 0);
+  gsize rbsp_size = gst_buffer_get_size (nalu->rbsp_buf);
+  g_assert_cmpuint (fu_pyl_max_size, <, rbsp_size);
+
+  // Split the NALU into several FU
+  for (gsize offset = 0; offset < rbsp_size; offset += fu_pyl_max_size) {
+    gsize remaining = rbsp_size - offset;
+    gboolean last = remaining <= fu_pyl_max_size;
+    gsize size = last ? remaining : fu_pyl_max_size;
+
+    GstBuffer *fu_buf =
+        gst_rtp_h266_pay_extract_fu (rtph266pay, nalu, offset, size, last);
+    if (!fu_buf)
+      goto error;
+    GST_DEBUG_OBJECT (rtph266pay, "FU [%lu, %lu] %" GST_PTR_FORMAT,
+        offset, offset + size, fu_buf);
+    gst_buffer_list_add (out_buflist, fu_buf);
+  }
+
+  GST_DEBUG_OBJECT (rtph266pay, "Pushing FU GstBufferList %" GST_PTR_FORMAT,
+      out_buflist);
+  return gst_rtp_base_payload_push_list (rtpbasepay, out_buflist);
+error:
+  gst_buffer_list_unref (out_buflist);
+  return GST_FLOW_ERROR;
+}
+
+static gboolean
+gst_rtp_h266_pay_up_fits_in_mtu (const GstRtpH266Pay * rtph266pay,
+    const Nalu * nalu)
+{
+  guint mtu = GST_RTP_BASE_PAYLOAD_MTU (rtph266pay);
+  guint payload_size = nalu->size;
+  // TODO: Consider DONL
+  gboolean fits = gst_rtp_buffer_calc_packet_len (payload_size, 0, 0) <= mtu;
+
+  if (!fits) {
+    GST_DEBUG_OBJECT (rtph266pay,
+        "NALU does not fit into %u MTU: %" GST_PTR_FORMAT, mtu, nalu->nalu_buf);
+  }
+  return fits;
+}
+
+static GstBuffer *
+gst_rtp_h266_pay_extract_fu (GstRtpH266Pay * rtph266pay, const Nalu * nalu,
+    gsize offset, gsize size, gboolean last)
+{
+  GstRTPBasePayload *rtpbasepay = GST_RTP_BASE_PAYLOAD (rtph266pay);
+  GstRTPBuffer rtp = GST_RTP_BUFFER_INIT;
+  gboolean au_end = last && nalu->au_end;
+  gboolean first = (offset == 0);
+  GstBuffer *fu_pyl_buf;
+  GstBuffer *fu_buf;
+  guint8 *fu_hdr;
+
+  fu_pyl_buf = gst_buffer_copy_region (nalu->rbsp_buf, GST_BUFFER_COPY_MEMORY,
+      offset, size);
+  if (!fu_pyl_buf)
+    return NULL;
+
+  // Allocate just the RTP header + PayloadHdr + FU header. We'll add the buffer
+  // payload later. This way we avoid unnecessary copies
+  fu_buf =
+      gst_rtp_base_payload_allocate_output_buffer (rtpbasepay, FU_HDR_LEN, 0,
+      0);
+  if (!gst_rtp_buffer_map (fu_buf, GST_MAP_WRITE, &rtp))
+    goto error_unref;
+
+  // Copy required buffer metadata
+  gst_buffer_copy_into (fu_buf, nalu->nalu_buf,
```

Comment:
Handle error? assert TRUE?

Reply:
We're only coping the flags and the timestamps; I think it's not that critical to abort. And because we'll already receive a warning in case of failure, I don't think it's worth it to deal with the returned value here.

Reply:
nitpick: `g_assert (gst_rtp_buffer_map (fu_buf, GST_MAP_WRITE, &rtp))`

Reply:
You shouldn't put logic into `g_assert()`, because it might not be executed:
```c
#ifdef G_DISABLE_ASSERT
#define g_assert(expr)                  G_STMT_START { (void) 0; } G_STMT_END
```

Reply:
Oh, then it's fine. Sorry, I did not know it.